apps/oc_chef_authz/src/oc_chef_policy_group_revision_association.erl

```diff
@@ -0,0 +1,207 @@
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil; fill-column: 92 -*-
+%% ex: ts=4 sw=4 et
+%% @author Daniel DeLeo <dan@chef.io>
+%% Copyright 2015 Chef Software, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+-module(oc_chef_policy_group_revision_association).
+
+-include("../../include/oc_chef_types.hrl").
+
+-export([find_policy_revision_by_orgid_name_group_name/1,
+         insert_association/1,
+         update_association/1,
+         delete_association/1,
+
+         id/1,
+         fields_for_fetch/1,
+         fields_for_update/1,
+         find_query/0,
+         create_query/0,
+         update_query/0,
+         delete_query/0,
+         record_fields/0,
+         flatten/1,
+         is_indexed/0,
+         ejson_for_indexing/2]).
+
+find_policy_revision_by_orgid_name_group_name(Record) ->
+    case chef_sql:fetch_object(
+      chef_object:fields_for_fetch(Record),
+      element(1, Record),
+      chef_object:find_query(Record),
+      chef_object:record_fields(Record)
+      ) of
+        {ok, #oc_chef_policy_group_revision_association{} = Assoc} ->
+          Decompressed = decompress_record(Assoc),
+          {ok, Decompressed};
+        Any -> Any
+    end.
+
+insert_association(#oc_chef_policy_group_revision_association{} = Record) ->
+    ok = ensure_policy_exists(Record),
+    ok = ensure_policy_group_exists(Record),
+    ok = ensure_policy_revision_exists(Record),
```

Comment:
Can these constraints be enforced with FKs instead of having to do that work here?